public/app/features/dashboard/utils/dashboard.ts

```diff
@@ -91,3 +91,66 @@ export function getCopiedPanelPlugin(): (PanelPluginMeta & PanelPluginInfo) | un
 
   return undefined;
 }
+
+export function initLocalStorageDashboard(uid: string | undefined) {
+  // set local storage for panel edit page
+  initLocalStoragePanelEdit(uid);
+}
+
+export function initLocalStoragePanelEdit(dashUid: string | undefined) {
+  // Implement logic to handle last used datasource in panel edit page
+  initLastUsedDatasourceLocalStorage(dashUid);
+}
+
+type LastUsedDatasource =
+  | {
+      dashboardUid: string;
+      datasourceUid: string;
+    }
+  | undefined;
+
+const PANEL_EDIT_LAST_USED_DATASOURCE = 'grafana.dashboards.panelEdit.lastUsedDatasource';
+
+function initLastUsedDatasourceLocalStorage(dashUid: string | undefined) {
+  const PANEL_EDIT_LAST_USED_DATASOURCE = 'grafana.dashboards.panelEdit.lastUsedDatasource';
```

Comment:
why do we also have this const here?

Reply:
good catch! it was left over of the exploratory code I did before 👍🏾